sentry_sdk/integrations/starlite.py

```diff
@@ -114,51 +115,50 @@ def enable_span_for_middleware(middleware: "Middleware") -> "Middleware":
         old_call = middleware.__call__
 
     async def _create_span_call(
-        self: "MiddlewareProtocol", scope: "Scope", receive: "Receive", send: "Send"
+        self: "MiddlewareProtocol",
+        scope: "StarliteScope",
+        receive: "Receive",
+        send: "Send",
     ) -> None:
-        hub = Hub.current
-        integration = hub.get_integration(StarliteIntegration)
-        if integration is not None:
-            middleware_name = self.__class__.__name__
-            with hub.start_span(
-                op=OP.MIDDLEWARE_STARLITE, description=middleware_name
-            ) as middleware_span:
-                middleware_span.set_tag("starlite.middleware_name", middleware_name)
-
-                # Creating spans for the "receive" callback
-                async def _sentry_receive(
-                    *args: "Any", **kwargs: "Any"
-                ) -> "Union[HTTPReceiveMessage, WebSocketReceiveMessage]":
-                    hub = Hub.current
-                    with hub.start_span(
-                        op=OP.MIDDLEWARE_STARLITE_RECEIVE,
-                        description=getattr(receive, "__qualname__", str(receive)),
-                    ) as span:
-                        span.set_tag("starlite.middleware_name", middleware_name)
-                        return await receive(*args, **kwargs)
-
-                receive_name = getattr(receive, "__name__", str(receive))
-                receive_patched = receive_name == "_sentry_receive"
-                new_receive = _sentry_receive if not receive_patched else receive
-
-                # Creating spans for the "send" callback
-                async def _sentry_send(message: "Message") -> None:
-                    hub = Hub.current
-                    with hub.start_span(
-                        op=OP.MIDDLEWARE_STARLITE_SEND,
-                        description=getattr(send, "__qualname__", str(send)),
-                    ) as span:
-                        span.set_tag("starlite.middleware_name", middleware_name)
-                        return await send(message)
-
-                send_name = getattr(send, "__name__", str(send))
-                send_patched = send_name == "_sentry_send"
-                new_send = _sentry_send if not send_patched else send
-
-                return await old_call(self, scope, new_receive, new_send)
-        else:
+        if sentry_sdk.get_client().get_integration(StarliteIntegration) is None:
```

Comment:
We can use the @encure_... decorator here, no?

Reply:
I'd do this after we've fixed it and we're sure it works as intended, then we can go over the async integrations and add it.